release-20.2: delegate: gate estimated_row_count on SHOW TABLES behind cluster setting

otan opened the pull request:
Refs: https://github.com/cockroachdb/cockroach/issues/58189

Release note (sql change): Introduced a cluster setting
`sql.show_tables.estimated_row_count.enabled`, which defaults to true. If
false, estimated_row_count will not display on SHOW TABLES which
improves performance.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/59776)
<!-- Reviewable:end -->

rafiss commented on pkg/sql/logictest/testdata/logic_test/show_table:
```diff
@@ -0,0 +1,17 @@
+# LogicTest: local
+
+statement ok
+SET CLUSTER SETTING sql.show_tables.show_estimated_row_count = false
```
i guess it has to be called `sql.show_tables.estimated_row_count.enabled` ? weird naming ergonomics..

otan committed:
delegate: gate estimated_row_count on SHOW TABLES behind cluster setting

Release note (sql change): Introduced a cluster setting
`sql.show_tables.estimated_row_count.enabled`, which defaults to true. If
false, estimated_row_count will not display on SHOW TABLES which
improves performance.

rafiss reviewed (approved):
lgtm! did we decide that this would be the solution for 21.1 too?

otan commented:
> lgtm! did we decide that this would be the solution for 21.1 too?

nah something else probably
